### DISCUSSION TITLE
Support for Qt?

### DISCUSSION THREAD

**Discussion by dourouc05** [carbon-language/carbon-lang#1866] [Language design]
Hi all,

Qt is probably one major library that makes C++ relevant in many industries. However, it is really hard to use in other languages: even in Python, PyQt and PySide/Qt for Python do not really provide Pythonic interfaces for everything. Most languages focus on bringing access to Qt Quick / QML rather than Qt.

Would Carbon bring benefits to projects like Qt and its users? The main problems would be the use of a meta-compiler ([moc](https://doc.qt.io/qt-6/moc.html)), which generates a lot of codes so that Qt can do its magic. This is not standard C++, although it can be implemented on the compiler-side (like [moc-ng](https://github.com/woboq/moc-ng)).

**Comment by jonmeow** [comment]
Carbon is still in a very early state, and we aren't building real porting plans for any C++ ecosystem library (the closest we come is trying to figure out a [couple low-complexity proof-of-concepts](https://github.com/carbon-language/carbon-lang/blob/trunk/docs/project/roadmap.md#example-ports-of-c-libraries-to-carbon-100-of-woff2-99-of-re2)). However, it should probably be expected that any system which relies on generating or parsing C++ code is outside the scope of what we can address with a C++ -> Carbon migration tooling. A typical solution in this scenario, lacking Carbon-specific support in moc, might be to build a thin C++ wrapper that calls into Carbon code.
